backend/fetchers/category_payments.rb

```diff
@@ -0,0 +1,15 @@
+class DataFetcher
+  class CategoryPayments
+    def self.run!
+      ActiveRecord::Base.connection.execute <<-QUERY
+	INSERT INTO category_payments (payer_id, text, code, amount)
+	SELECT  payer_id, 
+	  COALESCE(text, 'Not stated'), p.code, sum(amount)
+	FROM (payments p left outer join payment_codes c
+		      on p.code = c.code)
+	GROUP BY payer_id, text, p.code
```

Comment:
right.  having both text and code in the GROUP BY lets you put them both in the SELECT list.  Some systems will let you put non-grouped columns in the select list, but its not standard.